File: backend/app/schemas/nomenclatura.py
from datetime import datetime

from pydantic import BaseModel


class NomenclaturaPlaguicidaItem(BaseModel):
    ingrediente_key: str
    agrupador: str | None
    codigo: str | None
    fecha_mod: datetime | None


class NomenclaturaPlaguicidaUpdate(BaseModel):
    agrupador: str
    codigo: str | None = None


class PaginaNomenclaturaPlaguicidas(BaseModel):
    total: int
    pagina: int
    tamano_pagina: int
    filas: list[NomenclaturaPlaguicidaItem]


class AgrupadorNutrienteItem(BaseModel):
    nombre_key: str
    producto_agrupado: str | None
    codigo: str | None
    fecha_mod: datetime | None


class AgrupadorNutrienteUpdate(BaseModel):
    producto_agrupado: str
    codigo: str | None = None


class PaginaAgrupadorNutrientes(BaseModel):
    total: int
    pagina: int
    tamano_pagina: int
    filas: list[AgrupadorNutrienteItem]


# --- Sección "sin agrupador" (ex pantalla de Excepciones, consolidada
# dentro de Nomenclatura — ver admin_nomenclatura.py) ---


class ResumenSinAgrupadorPlaguicida(BaseModel):
    ingrediente_key: str
    ingrediente_ejemplo: str | None
    transacciones: int
    cantidad_total: float
    cif_usd_total: float
    posible_error_captura: bool


class DetalleSinAgrupadorPlaguicida(BaseModel):
    recibointerno: str | None
    ingrediente_act: str | None
    ingrediente_key: str | None
    producto: str | None


class PaginaDetalleSinAgrupadorPlaguicidas(BaseModel):
    total: int
    pagina: int
    tamano_pagina: int
    filas: list[DetalleSinAgrupadorPlaguicida]


class SinAgrupadorPlaguicidasResponse(BaseModel):
    resumen: list[ResumenSinAgrupadorPlaguicida]
    detalle: PaginaDetalleSinAgrupadorPlaguicidas


class ResumenSinAgrupadorNutriente(BaseModel):
    nombre_key: str
    nombre_ejemplo: str | None
    transacciones: int
    cif_dolares_total: float
    posible_error_captura: bool


class DetalleSinAgrupadorNutriente(BaseModel):
    no_licencia: str | None
    nombre_comercial: str | None
    nombre_key: str | None


class PaginaDetalleSinAgrupadorNutrientes(BaseModel):
    total: int
    pagina: int
    tamano_pagina: int
    filas: list[DetalleSinAgrupadorNutriente]


class SinAgrupadorNutrientesResponse(BaseModel):
    resumen: list[ResumenSinAgrupadorNutriente]
    detalle: PaginaDetalleSinAgrupadorNutrientes

File: frontend/src/types/nomenclatura.ts
export interface NomenclaturaPlaguicidaItem {
  ingrediente_key: string;
  agrupador: string | null;
  codigo: string | null;
  fecha_mod: string | null;
}

export interface PaginaNomenclaturaPlaguicidas {
  total: number;
  pagina: number;
  tamano_pagina: number;
  filas: NomenclaturaPlaguicidaItem[];
}

export interface AgrupadorNutrienteItem {
  nombre_key: string;
  producto_agrupado: string | null;
  codigo: string | null;
  fecha_mod: string | null;
}

export interface PaginaAgrupadorNutrientes {
  total: number;
  pagina: number;
  tamano_pagina: number;
  filas: AgrupadorNutrienteItem[];
}

// --- Sección "sin agrupador" (ex pantalla de Excepciones) ---

export interface ResumenSinAgrupadorPlaguicida {
  ingrediente_key: string;
  ingrediente_ejemplo: string | null;
  transacciones: number;
  cantidad_total: number;
  cif_usd_total: number;
  posible_error_captura: boolean;
}

export interface DetalleSinAgrupadorPlaguicida {
  recibointerno: string | null;
  ingrediente_act: string | null;
  ingrediente_key: string | null;
  producto: string | null;
}

export interface PaginaDetalleSinAgrupadorPlaguicidas {
  total: number;
  pagina: number;
  tamano_pagina: number;
  filas: DetalleSinAgrupadorPlaguicida[];
}

export interface SinAgrupadorPlaguicidasResponse {
  resumen: ResumenSinAgrupadorPlaguicida[];
  detalle: PaginaDetalleSinAgrupadorPlaguicidas;
}

export interface ResumenSinAgrupadorNutriente {
  nombre_key: string;
  nombre_ejemplo: string | null;
  transacciones: number;
  cif_dolares_total: number;
  posible_error_captura: boolean;
}

export interface DetalleSinAgrupadorNutriente {
  no_licencia: string | null;
  nombre_comercial: string | null;
  nombre_key: string | null;
}

export interface PaginaDetalleSinAgrupadorNutrientes {
  total: number;
  pagina: number;
  tamano_pagina: number;
  filas: DetalleSinAgrupadorNutriente[];
}

export interface SinAgrupadorNutrientesResponse {
  resumen: ResumenSinAgrupadorNutriente[];
  detalle: PaginaDetalleSinAgrupadorNutrientes;
}
